internal/venice-common/src/main/java/com/linkedin/venice/throttle/EventThrottler.java

```diff
@@ -168,6 +168,9 @@ public EventThrottler(
     Validate.notNull(time);
     Validate.notNull(throttlingStrategy);
     this.time = time;
+    if (maxRatePerSecondProvider.getAsLong() == 0) {
```

Comment:
We had a checker below: 
    if (maxRatePerSecond >= 0) {
      initialize(maxRatePerSecond);
    }
Maybe combine it with maxRatePerSecondProvider.getAsLong() == 0 here